cmake/helpers.cmake

```diff
@@ -104,3 +108,31 @@ function(add_umf_library)
     add_umf_target_compile_options(${ARG_NAME})
     add_umf_target_link_options(${ARG_NAME})
 endfunction()
+
+# Add sanitizer ${flag}, if it is supported, for both C and C++ compiler
+macro(add_sanitizer_flag flag)
+	set(SAVED_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
+	set(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES} -fsanitize=${flag}")
+
+	if(${flag} STREQUAL "address")
+		set(check_name "HAS_ASAN")
+	elseif(${flag} STREQUAL "undefined")
+		set(check_name "HAS_UBSAN")
+	endif()
+
+	check_c_compiler_flag("-fsanitize=${flag}" "C_${check_name}")
```

Comment:
make sure that this macro support g++, clang and cl (windows)

Reply:
I've configured `add_sanitizer_flag` macro to trigger an error when any of the sanitizers is enabled but not supported by the compiler.